code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/file/docker/OpendDockerComposeFile.xtend

```diff
@@ -0,0 +1,242 @@
+        } else {
+            createDefaultNetwork
+        }
+    }
+
+    /*
+     * Add or replace a docker compose part to the docker compose file based on the name of the
+     * operation node.The configuration of the docker compose part is created by an intermediate
+     * deployment aspect.
+     */
+    def addOrReplaceDockerComposePart(String name, String dockerComposeServicePart) {
+        this.addOrReplaceToMap(name,
+            Util::adjustIndentations('  ', dockerComposeServicePart) + '\n'
+        )
+    }
+
+    /*
+     * Add or replace a docker compose part to the docker compose file based on the name of the
+     * operation node. The configuration of the docker compose part is created by the docker compose
+     * code generation module itself.
+     *
+     * Note: the docker compose part is done separately for intermediate container and intermediate
+     * infrastructure nodes.
+     */
+    def addOrReplaceDockerComposePart(IntermediateOperationNode node) {
+        if (IntermediateInfrastructureNode.isInstance(node)) {
+            addOrReplaceToMap(node.name,
+                getServiceFromInfrastructureNode(node as IntermediateInfrastructureNode) + "\n"
+            )
+        } else {
+            addOrReplaceToMap(node.name,
+                getServiceFromContainer(node as IntermediateContainer) +"\n"
+            )
+        }
+    }
+
+    /*
+     * Create a default network for the docker compose file. This is needed because LEMMA's
+     * operation modeling language does not support network configurations at the moment.
+     */
+    private def createDefaultNetwork() {
+            var network = new Network
+            network.driver = "bridge"
+            dockerComposeNetworkParts.add(
+                DockerComposeTemplate::buildDockerComposeNetwork("default-network", network)
+            )
+    }
+
+    /*
+     * Add or replace the docker compose parts based the the name of the intermediate operation
+     * node.
+     */
+    private def addOrReplaceToMap(String name, String dockerComposeServicePart) {
+        if (dockerComposeServiceParts.containsKey(name.toLowerCase)) {
+            dockerComposeServiceParts.replace(name.toLowerCase, dockerComposeServicePart)
+        } else {
+            dockerComposeServiceParts.put(name.toLowerCase, dockerComposeServicePart)
+        }
+    }
+
+    /*
+     * Create a POJO [Service] from  a container
+     */
+    private def String getServiceFromContainer(IntermediateContainer container) {
+        val service = new Service
+
+        service.build = Util::buildPathFromQualifiedName(
+            container.deployedServices.get(0).qualifiedName
+        )
+
+        service.image = container.name
+        service.container_name = container.name
+        container.endpoints
+
+        container.endpoints.forEach[endpoint |
+            endpoint.addresses.forEach[address |
+               service.ports.add(Util::getPortFromAddress(address) + ':'
```

Comment:
Please introduce a variable to prevent the duplicate call to getPortFromAddress(), which will supposedly result in the same value anyways given the same `address` argument:
```
val port = Util::getPortFromAddress(address)
service.ports.add('''«port»:«port»''')
```
also increases readability and comprehensibility.